### PR TITLE
types: add Symbol

### LINKED ISSUE #133: Add $.Symbol
In https://github.com/sanctuary-js/sanctuary/issues/381 I mentioned that `R.pluck` works with Arrays and Number indices as opposed to just StrMaps and String keys.

The signature of `S.pluck` _could_ be:
```hs
type Natural = PositiveInteger | 0

type AccessibleKey = String | Natural | Symbol

pluck :: (Accessible a, Functor f) => AccessibleKey -> f a -> f b
```

This would require `$.Symbol` (and possibly `$.Natural`).

### PR DESCRIPTION
**Pull request by gabejohnson**
Closes #133

### CODE REVIEW

**Comment by gabejohnson** [comment]
@davidchambers will this change require adding Symbol to sanctuary-type-classes as well?

**Comment by davidchambers** [comment]
> @davidchambers will this change require adding Symbol to sanctuary-type-classes as well?

I don't think so. What do you have in mind?

File: index.js [hunk 1]
```diff
@@ -713,6 +713,12 @@
   //. Type comprising every primitive String value.
   var String_ = NullaryTypeWithUrl('String', typeofEq('string'));
 
+
```
Comment (davidchambers) [inline index.js]: Nit: Unwanted line.

File: index.js [hunk 2]
```diff
@@ -713,6 +713,12 @@
...
+  //# Symbol :: Type
+  //.
+  //. Type comprising every primitive Symbol value.
```
Comment (davidchambers) [inline index.js]: Is it necessary to use "primitive" here?

File: index.js [hunk 3]
```diff
@@ -713,6 +713,12 @@
...
+  var Symbol_ = NullaryTypeWithUrl('Symbol', typeofEq('symbol'));
```
Comment (davidchambers) [inline index.js]: Is there a reason to favour `typeofEq('symbol')` over `typeEq('Symbol')`, or are the expressions equivalent?

File: index.js [hunk 4]
```diff
@@ -793,6 +800,7 @@
     RegExp_,
     StrMap(Unknown),
     String_,
+    Symbol,
```
Comment (davidchambers) [inline index.js]: `s/Symbol/Symbol_/`

File: index.js (hunk 2 above)
Comment (gabejohnson) [inline index.js]: An artifact of copying. There is no boxed version of `Symbol` thus no need to differentiate it.

Though it is a primitive.

File: index.js (hunk 3 above)
Comment (gabejohnson) [inline index.js]: They should be equivalent. I would think the `typeofEq` would be slightly faster. I don't know if there are any benchmarks though.

**Comment by gabejohnson** [comment]
Updated

File: index.js (hunk 3 above)
Comment (davidchambers) [inline index.js]: :thumbsup:

**Pull request merged**